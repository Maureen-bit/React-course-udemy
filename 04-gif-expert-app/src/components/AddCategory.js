import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const changeText = (event) => {
        setInputValue(event.target.value);
    };

    const onSubmitForm = (event) =>{
        /* Evitar comportamiento pot defecto del formulario de cambiar de ruta 
        ( de http://localhost:3000 a http://localhost:3000/?) y hacer post de la informacion del formulario  */
        event.preventDefault();
        if(inputValue.trim().length > 2){
        /* Se llama como un call back para tener como referencia el estado anterior, es decir el valor
        de categories en el useState */
        setCategories((categories) => [ inputValue, ...categories ]);
        /* Para que al dar enter limpie el input y quede vacio para escribir otra cosa */
        setInputValue('');
        console.log('Submit hecho!')
    }
    }

  return (
      <form onSubmit={(event) => onSubmitForm(event)} >
        <p>{inputValue}</p>
        <input 
            type="text"
            value={inputValue}
            onChange={ (event) => changeText(event)}
        />
    </form>
  )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };