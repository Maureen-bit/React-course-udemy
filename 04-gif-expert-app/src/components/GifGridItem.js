import React from 'react';
import PropTypes from 'prop-types';
import 'animate.css';

export const GifGridItem = ( {title, url} ) => {
  return (
    <div className='card animate__animated animate__heartBeat'>
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
  )
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};