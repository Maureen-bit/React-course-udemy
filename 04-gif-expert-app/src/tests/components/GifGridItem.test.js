import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas unitarias componente <GifGridItem />", () => {
    let title = "Gif One";
    let url = "www.gifone.com";
    const view = shallow(<GifGridItem title={title} url={url} />);
    test('Debe de mostrar el componente correctamente', () => {
      expect(view).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => {
      const p = view.find('p');
      expect(p.text().trim()).toBe(title);
    });

    test('Debe tener una imagen con alt y src de los props', () => {
      const image = view.find('img');
      // acceder a las propiedades de un elemento .props().propiedadBuscada o .props('propiedadBuscada')
      expect(image.props().src).toBe(url);
      expect(image.props().alt).toBe(title);
    });
    
    test('Debe tener el className: animate__heartBeat en el div', () => {
      const div = view.find('div');
      const classDiv  = div.props().className; // o tambi√©n: div.props('className');
      /* Forma uno */
      expect(classDiv).toContain('animate__heartBeat');
      /* Forma dos */
      expect(classDiv.includes('animate__heartBeat')).toBe(true);
    });
})