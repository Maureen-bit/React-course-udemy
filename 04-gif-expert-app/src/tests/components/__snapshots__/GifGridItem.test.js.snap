// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias componente <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__heartBeat"
>
  <img
    alt="Gif One"
    src="www.gifone.com"
  />
  <p>
    Gif One
  </p>
</div>
`;
